dm/worker/subtask.go

```diff
@@ -236,18 +236,16 @@ func (st *SubTask) callCurrCancel() {
 func (st *SubTask) fetchResult(pr chan pb.ProcessResult) {
 	defer st.wg.Done()
 
-	st.RLock()
-	ctx := st.currCtx
-	st.RUnlock()
-
 	select {
-	case <-ctx.Done():
+	case <-st.ctx.Done():
```

Comment:
Should we need to fix this in release-1.0? cc @lichunzhu

Reply:
I think it's not correct. We will start multi `fetchResult` in the end. `fetchResult` should be ended when one turn is finished.

Reply:
but we still waiting for `case result := <-pr:`, is that enough? @lichunzhu

Reply:
It's enough, and we mush assure `pr` will get a result in the future.